Wikipedia/Code/MWKImage.m

```diff
@@ -272,90 +218,34 @@ - (BOOL)isEqual:(id)object {
 }
 
 - (BOOL)isEqualToImage:(MWKImage*)image {
-    return self == image || [self.fileName isEqualToString:image.fileName];
+    return self == image || [[self.sourceURLString wmf_schemelessURL] isEqualToString:[image.sourceURLString wmf_schemelessURL]];
 }
 
 - (NSUInteger)hash {
     return [self.fileName hash];
 }
 
+- (BOOL)isPathEqualToImage:(MWKImage*)image {
+    return [[[self.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent] isEqualToString:[[image.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent]];
+}
+
 - (BOOL)isVariantOfImage:(MWKImage*)otherImage {
     // !!!: this might not be reliable due to underscore, percent encodings, and other unknowns w/ image filenames
     return [self.fileNameNoSizePrefix isEqualToString:otherImage.fileNameNoSizePrefix]
-           && ![self isEqualToImage:otherImage];
+           && [self isPathEqualToImage:otherImage];
```

Comment:
@montehurd in those cases, the image filename extraction APIs should return the second to the last path component, which is the actual filename, not the thumbnail filename